textalyze.js

```diff
@@ -29,7 +31,7 @@ function itemCounts(array) {
  * @returns {Array} The Array containing the characters from the string
  */
 function arrayFrom(string) {
```

Comment:
Check out [Array.from](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from).  I'm not saying this is what you should use, but compare its semantics to the semantics of your `arrayFrom` function.